docs/proposals/proposal-workflow-authentication.md

```diff
@@ -0,0 +1,69 @@
+# Proposal for Workflow Authentication
+
+## Proposal summary
+
+**For workflow-to-workflow-service authentication, use:**
+
+* **for jobs: one-time per-job tokens**, i.e. a single one-time token for
+  every job;
+* **for the user/interactive CLI: timed tokens**.
+
+Per-job tokens would require one of the following (or another alternative):
+
+  * another file per job, which is not ideal as it adds to the file count;
+  * a shared combined file containing tokens for all jobs per workflow, also
+    not ideal as it means that jobs would needlessly know about other jobs'
+    tokens;
```

Comment:
A dumped hash table is not so different to the above, because job has access to other user files in the file system any way.